bookshelf/tests/api_v3/test_cloud.py

```diff
@@ -0,0 +1,339 @@
+                        'public_key_file': '<path-to-corresponding-public-key>'
+                    }
+                }
+            },
+            config=raw_config
+        )
+
+        keys = credentials["ssh_keys"]["gce"]
+
+        # If specified, attempt to use the service account credentials from
+        # `acceptance.yml` rather than the default authentication method.
+        service_account_creds = (
+            raw_config.get('gce', {}).get('gce_credentials', {}))
+
+        self.config = GCEConfiguration(
+            credentials_private_key=(
+                service_account_creds.get('private_key', '')),
+            credentials_email=service_account_creds.get('client_email', ''),
+            public_key_filename=keys["public_key_file"],
+            private_key_filename=keys["private_key_file"],
+            project=credentials["gce"]["project"],
+            machine_type="n1-standard-1",
+            image_basename="gce-test-image",
+            username="gce-username-xyz",
+            description="gce-test-description",
+            instance_name="gce-test-instance",
+            base_image_prefix='ubuntu-1404',
+            base_image_project='ubuntu-os-cloud'
+        ).serialize()
+        self.distribution = Distribution.UBUNTU1404
+        self.region = 'us-central1-f'
+        self.instance_factory = GCEInstance
+
+
+class EC2Tests(unittest.TestCase, CloudInstanceTestMixin):
+    """
+    Tests for EC2.
+    """
+
+    def setUp(self):
+        super(EC2Tests, self).setUp()
+
+        credentials = _get_yaml_config(
+            {
+                'aws': {
+                    'access_key': '<AWS-ACCESS-KEY>',
+                    'secret_access_token': '<AWS-SECRET-ACCESS-TOKEN>',
+                    'keyname': '<aws-us-west-2-ssh-keypair-name>',
+                },
+                'ssh_keys': {
+                    'aws': {
+                        'private_key_file': '<path-to-private-key-for-keypair>'
+                    }
+                }
+            }
+        )
+
+        self.config = EC2Configuration(
+            credentials=EC2Credentials(
+                access_key_id=credentials["aws"]["access_key"],
+                secret_access_key=credentials["aws"]["secret_access_token"]
+            ),
+            username='centos',
+            disk_name='/dev/sda1',
+            disk_size='48',
+            instance_name='ec2-test-instance',
+            tags={'name': 'test-instance-with-tags'},
+            image_description='ec2-test-description',
+            image_basename='ec2-image-basename',
+            ami='ami-d440a6e7',
+            key_filename=credentials["ssh_keys"]["aws"]["private_key_file"],
+            key_pair=credentials["aws"]["keyname"],
+            instance_type='t2.medium',
+            security_groups=['ssh']
+        ).serialize()
+        self.distribution = Distribution.CENTOS7
+        self.region = 'us-west-2'
```

Comment:
the above doesn't work,
but the ansible ec2-ami  plugin actually shows that it might be easier than we think
https://github.com/openshift/openshift-ansible/blob/master/playbooks/aws/openshift-cluster/library/ec2_ami_find.py#L149